### PR TITLE
chore(acl_family): add allcomands and nocommands

### LINKED ISSUE #3458: add `allcomands` and `nocommands` support for ACL's
* allcommands is syntactic sugar for `+@all`
* nocommands is syntactic sugar for `-@all`

### PR DESCRIPTION
**Pull request by Lakshyadevelops**
Fixes #3458 
TESTED =  
I did not add tests as there was no test earlier for the same but it can be added.

### CODE REVIEW

File: src/server/acl/acl_family.cc [hunk 1]
```diff
@@ -990,6 +990,22 @@ using OptCat = std::optional<uint32_t>;
 // bool == true if +
 // bool == false if -
 std::pair<OptCat, bool> AclFamily::MaybeParseAclCategory(std::string_view command) const {
+  if (absl::EqualsIgnoreCase(command, "ALLCOMMANDS")) {
+    auto res = cat_table_.find("ALL");
```
Comment (kostasrim) [inline src/server/acl/acl_family.cc:994]: `ALL` will always exist in `cat_table_` so you don't really need to call `find` and compare that `cat_table_.end()`. 

You can just use `cat_table_["ALL"]` :)

**Review by kostasrim — COMMENTED** [review]
Hi @Lakshyadevelops,

Thank you so much for your contribution! Would you mind also adding a unit test ? 

I also left a very small nit :)

File: src/server/acl/acl_family.cc (hunk 1 above)
Comment (Lakshyadevelops) [reply src/server/acl/acl_family.cc]: Yeah, you are right about comparing to ```cat_table_.end()```

I tried to used operator[] but it was failing due to type mismatch and even after casting to ```std::string_view``` it was failing due to -fpermissive (this arguement discards qualifier)  so finally used ```.at()```

**Review by kostasrim — APPROVED** [review]
LGTM!

**Pull request merged**